addons/website/static/src/js/text_processing.js

```diff
@@ -0,0 +1,345 @@
+        // TODO: Improve this.
+        let lastNodeInLine = false;
+        if (child.nextSibling && child.nextSibling.textContent) {
+            const range = document.createRange();
+            const lastCurrentText = selectAllTextNodes(child).at(-1);
+            range.setStart(lastCurrentText, lastCurrentText.length - 1);
+            range.setEnd(lastCurrentText, lastCurrentText.length);
+            // Get the "END" position of the last text node in current child.
+            const currentEnd = range.getBoundingClientRect()[isRTL(topTextEl) ? "left" : "right"];
+            const firstnextText = selectAllTextNodes(child.nextSibling)[0];
+            range.setStart(firstnextText, 0);
+            range.setEnd(firstnextText, 1);
+            // Get the "START" position of the first text node in the next
+            // sibling.
+            const nextStart = range.getBoundingClientRect()[isRTL(topTextEl) ? "right" : "left"];
+            // The next sibling starts before the end of the current node
+            // => Line break detected.
+            lastNodeInLine = nextStart + 1 < currentEnd;
+        }
+
+        // for each text line detected, we add the content as new
+        // line and adjust the line index accordingly.
+        textLines.map((node, i, {length}) => {
+            if (!lines[lineIndex]) {
+                lines[lineIndex] = [];
+            }
+            lines[lineIndex].push(node);
+            if (i !== length - 1 || lastNodeInLine) {
+                lineIndex++;
+            }
+        });
+    });
+    topTextEl.replaceChildren(...lines.map(textLine => {
+        // First we add text content to be able to build svg paths
+        // correctly (`<br/>` tags are excluded).
+        return nodeIsBR(textLine[0]) ? textLine[0] :
+            createHighlightContainer(textLine);
+    }));
+    // Build and set highlight svg.
+    [...topTextEl.querySelectorAll(".o_text_highlight_item")].forEach(container => {
+        container.append(drawTextHighlightSVG(container, highlightID));
+    });
+}
+
+/**
+ * Used to rollback the @see setTextHighlight behaviour.
+ *
+ * @param {HTMLElement} topTextEl
+ */
+export function removeTextHighlight(topTextEl) {
+    // Simply replace every `<span class="o_text_highlight_item">
+    // textNode1 [textNode2,...]<svg .../></span>` by `textNode1
+    // [textNode2,...]`.
+    [...topTextEl.querySelectorAll(".o_text_highlight_item")].forEach(unit => {
+        unit.replaceWith(...[...unit.childNodes].filter((node) => node.tagName !== "svg"));
+    });
+    // Prevents incorrect text lines detection on the next updates.
+    let child = topTextEl.firstElementChild;
+    while (child) {
+        let next = child.nextElementSibling;
+        // Merge identical elements.
+        if (next && next === child.nextSibling && child.cloneNode().isEqualNode(next.cloneNode())) {
+            child.replaceChildren(...child.childNodes, ...next.childNodes);
+            next.remove();
```

Comment:
I think in the case you're talking about, you have `<srong>Hello</strong><whitespace text node><strong>Hello</strong>` instead of `<srong>Hello</strong><strong>Hello</strong>` The whitespace text node will be ignored when rendering text highlights items (see: https://developer.mozilla.org/en-US/docs/Web/API/Document_Object_Model/Whitespace)... Code updated now to preserve them inside the `.o_text_highlight_item` elements :+1: